lab4/finite_automata.py

```diff
@@ -39,9 +40,38 @@ def read_from_file(self, fileName):
 
         return FiniteAutomata(Q, E, S, q0, F)
 
+    def is_dfa(self):
+        transition_list = []
+        for transition in self.S:
+            trans_tuple = (transition.start_state, transition.value)
+            if trans_tuple in transition_list:
+                return False
+            transition_list.append(trans_tuple)
+        return True
+
+    def verify_sequence(self, sequence):
+        # keep the current state: we start from the initial state
+        current_state = self.q0
+
+        # construct dictionary of transitions
```

Comment:
Maybe it would help constructing the dictionary of transitions from the very beginning (after reading the input) and not every time you want to test a new sequence